biojava-structure/src/test/java/org/biojava/nbio/structure/align/client/TestStructureName.java

```diff
@@ -115,72 +118,82 @@ public void testPrefixes() throws StructureException {
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 		sn = new StructureName("PDB:4hhb.A:1-50");
 		assertTrue(sn.isPdbId());
 		assertTrue(sn.getSource() == PDB);
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 		// Invalid strings work too, they just don't load
-		sn = new StructureName("PDB:x");
-		assertTrue(sn.isPdbId());
-		assertTrue(sn.getSource() == PDB);
-		assertEquals("x",sn.getPdbId());
+		try {
+			sn = new StructureName("PDB:x");
+			assertTrue(sn.isPdbId());
+			assertTrue(sn.getSource() == PDB);
+			assertEquals("x",sn.getPdbId());
+			fail("Current PDBId class does not allow malformed PdbId");
+		}catch (IllegalArgumentException e) {
+		}
 		// SCOP
 		sn = new StructureName("SCOP:d2gs2a_");
 		assertTrue(sn.isScopName());
 		assertTrue(sn.getSource() == SCOP);
-		assertEquals("2GS2",sn.getPdbId());
+		assertEquals(new PDBId("2GS2"),sn.getPdbId());
 		// CATH
 		sn = new StructureName("CATH:1qvrC03");
 		assertTrue(sn.isCathID());
 		assertTrue(sn.getSource() == CATH);
-		assertEquals("1QVR",sn.getPdbId());
+		assertEquals(new PDBId("1QVR"),sn.getPdbId());
 		// URL
 		sn = new StructureName("URL:http://www.rcsb.org/pdb/files/1B8G.pdb.gz");
 		assertTrue(sn.isURL());
 		assertTrue(sn.getSource() == URL);
-		assertEquals("1B8G",sn.getPdbId());
+		assertEquals(new PDBId("1B8G"),sn.getPdbId());
 		sn = new StructureName("URL:file:///4hhb.pdb");
 		assertTrue(sn.isURL());
 		assertTrue(sn.getSource() == URL);
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 		// File: expand home directory (invalid URL)
 		sn = new StructureName("FILE:~/4hhb.pdb");
 		assertTrue(sn.isFile());
 		assertTrue(sn.getSource() == FILE);
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 		// Relative file (invalid URL)
 		sn = new StructureName("file:4hhb.pdb");
 		assertTrue(sn.isFile());
 		assertTrue(sn.getSource() == FILE);
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 		// Absolute paths are valid URLs
 		sn = new StructureName("file:/4hhb_other.pdb");
 		assertTrue(sn.isURL());
 		assertTrue(sn.getSource() == URL);
-		assertEquals("4HHB",sn.getPdbId());
+		assertEquals(new PDBId("4HHB"),sn.getPdbId());
 
 		// ECOD
 		sn = new StructureName("e1lyw.1");
 		assertTrue(sn.isEcodDomain());
 		assertTrue(sn.getSource() == ECOD);
-		assertEquals("1LYW",sn.getPdbId());
+		assertEquals(new PDBId("1LYW"),sn.getPdbId());
 		// BIO
 		sn = new StructureName("BIO:2ehz:1");
 		assertTrue(sn.isBioAssembly());
 		assertTrue(sn.getSource() == BIO);
-		assertEquals("2EHZ",sn.getPdbId());
+		assertEquals(new PDBId("2EHZ"),sn.getPdbId());
 
 		// Invalid prefix
 		sn = new StructureName("XXX:2ehz");
 		assertTrue(sn.isPdbId());
 		assertTrue(sn.getSource() == PDB);
-		assertEquals("XXX:2ehz",sn.getPdbId());
+//		assertEquals("XXX:2ehz",sn.getPdbId());
```

Comment:
Because in the last line of StructureName.init(), the whole string `"XXX:2ehz"` is passed to the initFromPDB(String), not the suffix `2ehz` only. (although the function parameter is called `suffix` not `name`.
Of course `XXX:2ehz` does not match the PDBID format.
If we pass the suffix only, it will pass well, the PDBId object would be created, and there will be NO WARNING.
Any suggestions @sbliven ?

Reply:
UPDATE: After thinking over, I got to a conclusion.
If it is EXPLICITLY defined to be an UNKNOWN format, it should NOT be recognized. Hence, it defaults to null.